### PR TITLE
Adamo

### PR DESCRIPTION
**Pull request by adamoyoung**

### CODE REVIEW

File: src/data_loader.py [hunk 1]
```diff
@@ -211,46 +245,127 @@ def process(self):
...
-#         # One bond between atom u and v corresponds to two edges (u, v) and (v, u)
-#         feats.extend([btype, btype])
-#     return {'type': torch.tensor(feats).reshape(-1, 1).float()}
+    def __init__(self, purpose='training', use_ccri_negatives=True,  max_seq_len=None, stoi=None, **kwargs):
+        super(SelfiesCancerMolecules,self).__init__(purpose=purpose,use_ccri_negatives=use_ccri_negatives)
+        self.max_seq_len = max_seq_len
+        if purpose == "test":
+            # must load a previous stoi
+            assert stoi is not None
+        self.stoi = stoi
```
Comment (phil-fradkin) [inline src/data_loader.py:282]: What does stoi stand for?

File: src/data_loader.py [hunk 2]
```diff
@@ -261,10 +376,10 @@ def run_a_train_epoch(args, epoch, model, data_loader, loss_criterion, optimizer
 
 
 def split_data(dataset):
-    np.random.seed(1337)
-    assert type(dataset) == CancerMolecules
-    split = ScaffoldSplitter()
-    train, val, test = split.train_val_test_split(dataset, scaffold_func='smiles')
+    assert isinstance(dataset,CancerMolecules), type(dataset)
+    with np_temp_seed(13371337):
```
Comment (phil-fradkin) [inline src/data_loader.py:380]: Just a side comment I think declaring seed earlier works also (makes the next operations deterministic)

File: src/data_loader.py [hunk 3]
```diff
@@ -273,56 +388,64 @@ def run_an_eval_epoch(args, model, data_loader):
...
 
-    node_featurizer = CanonicalAtomFeaturizer(atom_data_field='h')
-    edge_featurizer = CanonicalBondFeaturizer(bond_data_field='h')
 
-    g = smiles_to_bigraph(
-        data.smiles[3],
-        node_featurizer=node_featurizer,
-        edge_featurizer=edge_featurizer
-    )
+# def main():
```
Comment (phil-fradkin) [inline src/data_loader.py:422]: I prefer to just delete dead code if anything we can go into commit history to retrieve it

**Review by phil-fradkin — APPROVED** [review]
Looks good I like the refactoring!

File: src/data_loader.py (hunk 1 above)
Comment (adamoyoung) [reply src/data_loader.py:282]: symbol to index! (I was following the SELFIES code)

File: src/data_loader.py (hunk 2 above)
Comment (adamoyoung) [reply src/data_loader.py:380]: yeah! I find the temp seeds to be useful because you can precisely control randomness of different parts of the code, so if you are comparing one version of code vs another you can change the order/number of random operations and still end up with the exact same behaviour. it basically just comes from uncertainty about which parts of the code are actually affecting the numpy/pytorch global random state (particularly if you rely on other libraries). we could even use a temp seed for the NN initialization for example.

Comment (phil-fradkin) [reply src/data_loader.py:380]: I see that is convenient!
